__tests__/app.test.js

```diff
@@ -217,4 +217,48 @@ describe('6. GET /api/articles - Retrieve all articles with comment count', () =
             })
         })
     })
+})
+describe('7. GET /api/articles/:article_id/comments', () => {
+    test('status: 200, should retrieve an array of comments from a given article id', () => {
+        let usernames = ['rogersop', 'icellusedkars', 'butter_bridge', 'lurker']
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+            expect(body).toBeInstanceOf(Object)
+            expect(body.comments).toHaveLength(11)
+            body.comments.forEach((comment) => {
+                expect(typeof(comment.comment_id)).toBe('number')
+                expect(typeof(comment.body)).toBe('string')
+                expect(typeof(comment.votes)).toBe('number')
+                expect(typeof(Date.parse(comment.created_at))).toBe('number')
+                expect(usernames.includes(comment.author)).toBe(true);
+            })
+            expect()
+        })
+    })
+    test('status: 404, should return a message when no comments found', () => {
+        return request(app)
+        .get('/api/articles/2/comments')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('Not Found')
+        })
```

Comment:
I'd argue that 404 is the incorrect error code in this case.

Because there is definitely an article with ID of 2, it's just that it doesn't have any comments **yet**.

Sending an empty array of comments would be acceptable in this case.

Have a think about how you could deal with this issue